### DISCUSSION TITLE
Game Crash with own coded Mod       The game crashed whilst initializing game Error: java.lang.RuntimeException: Could not execute entrypoint stage 'main' due to errors, provided by 'snowy'!

### DISCUSSION THREAD

**Discussion by dieser4DR** [FabricMC/fabric-api#2758] [Mod Dev Support]
---- Minecraft Crash Report ----
// Daisy, daisy...

Time: 2022-12-13 19:04:56
Description: Initializing game

java.lang.RuntimeException: Could not execute entrypoint stage 'main' due to errors, provided by 'snowy'!
	at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.lambda$invoke0$0(EntrypointUtils.java:51)
	at net.fabricmc.loader.impl.util.ExceptionUtil.gatherExceptions(ExceptionUtil.java:33)
	at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.invoke0(EntrypointUtils.java:49)
	at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.invoke(EntrypointUtils.java:35)
	at net.fabricmc.loader.impl.game.minecraft.Hooks.startClient(Hooks.java:52)
	at net.minecraft.class_310.<init>(class_310.java:459)
	at net.minecraft.client.main.Main.method_44604(Main.java:205)
	at net.minecraft.client.main.Main.main(Main.java:51)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:461)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)
Caused by: java.lang.NoClassDefFoundError: net/minecraft/client/gui/screen/Screen
	at java.base/java.lang.Class.forName0(Native Method)
	at java.base/java.lang.Class.forName(Class.java:467)
	at net.fabricmc.loader.impl.util.DefaultLanguageAdapter.create(DefaultLanguageAdapter.java:50)
	at net.fabricmc.loader.impl.entrypoint.EntrypointStorage$NewEntry.getOrCreate(EntrypointStorage.java:117)
	at net.fabricmc.loader.impl.entrypoint.EntrypointContainerImpl.getEntrypoint(EntrypointContainerImpl.java:53)
	at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.invoke0(EntrypointUtils.java:47)
	... 8 more
Caused by: java.lang.ClassNotFoundException: net.minecraft.client.gui.screen.Screen
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:520)
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.loadClass(KnotClassDelegate.java:226)
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:145)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:520)
	... 14 more

A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Render thread
Stacktrace:
	at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.lambda$invoke0$0(EntrypointUtils.java:51)
	at net.fabricmc.loader.impl.util.ExceptionUtil.gatherExceptions(ExceptionUtil.java:33)
	at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.invoke0(EntrypointUtils.java:49)
	at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.invoke(EntrypointUtils.java:35)
	at net.fabricmc.loader.impl.game.minecraft.Hooks.startClient(Hooks.java:52)
	at net.minecraft.class_310.<init>(class_310.java:459)

-- Initialization --
Details:
	Modules: 
		ADVAPI32.dll:Erweiterte Windows 32 Base-API:10.0.19041.2248 (WinBuild.160101.0800):Microsoft Corporation
		COMCTL32.dll:Bibliothek für Steuerelemente:6.10 (WinBuild.160101.0800):Microsoft Corporation
		CRYPT32.dll:Krypto-API32:10.0.19041.2248 (WinBuild.160101.0800):Microsoft Corporation
		CRYPTBASE.dll:Base cryptographic API DLL:10.0.19041.546 (WinBuild.160101.0800):Microsoft Corporation
		CRYPTSP.dll:Cryptographic Service Provider API:10.0.19041.546 (WinBuild.160101.0800):Microsoft Corporation
		DBGHELP.DLL:Windows Image Helper:10.0.19041.867 (WinBuild.160101.0800):Microsoft Corporation
		DNSAPI.dll:DNS-Client-API-DLL:10.0.19041.2248 (WinBuild.160101.0800):Microsoft Corporation
		GDI32.dll:GDI Client DLL:10.0.19041.2130 (WinBuild.160101.0800):Microsoft Corporation
		IMM32.DLL:Multi-User Windows IMM32 API Client DLL:10.0.19041.2193 (WinBuild.160101.0800):Microsoft Corporation
		IPHLPAPI.DLL:IP-Hilfs-API:10.0.19041.2248 (WinBuild.160101.0800):Microsoft Corporation
		KERNEL32.DLL:Client-DLL für Windows NT-Basis-API:10.0.19041.2248 (WinBuild.160101.0800):Microsoft Corporation
		KERNELBASE.dll:Client-DLL für Windows NT-Basis-API:10.0.19041.2248 (WinBuild.160101.0800):Microsoft Corporation
		MpOav.dll:IOfficeAntiVirus Module:4.18.2211.5 (WinBuild.160101.0800):Microsoft Corporation
		NLAapi.dll:Network Location Awareness 2:10.0.19041.2193 (WinBuild.160101.0800):Microsoft Corporation
		NSI.dll:NSI User-mode interface DLL:10.0.19041.610 (WinBuild.160101.0800):Microsoft Corporation
		NTASN1.dll:Microsoft ASN.1 API:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		Ole32.dll:Microsoft OLE für Windows:10.0.19041.2248 (WinBuild.160101.0800):Microsoft Corporation
		OleAut32.dll:OLEAUT32.DLL:10.0.19041.985 (WinBuild.160101.0800):Microsoft Corporation
		PSAPI.DLL:Process Status Helper:10.0.19041.546 (WinBuild.160101.0800):Microsoft Corporation
		Pdh.dll:Windows Unterstützungs-DLL für Leistungsdaten:10.0.19041.2248 (WinBuild.160101.0800):Microsoft Corporation
		RPCRT4.dll:Remoteprozeduraufruf-Laufzeitumgebung:10.0.19041.2248 (WinBuild.160101.0800):Microsoft Corporation
		SHCORE.dll:SHCORE:10.0.19041.2248 (WinBuild.160101.0800):Microsoft Corporation
		SHELL32.dll:Allgemeine Windows-Shell-DLL:10.0.19041.2248 (WinBuild.160101.0800):Microsoft Corporation
		USER32.dll:Client-DLL für Windows USER-API (mehrere Benutzer):10.0.19041.2248 (WinBuild.160101.0800):Microsoft Corporation
		USERENV.dll:Userenv:10.0.19041.2248 (WinBuild.160101.0800):Microsoft Corporation
		VCRUNTIME140.dll:Microsoft® C Runtime Library:14.29.30139.0 built by: vcwrkspc:Microsoft Corporation
		VERSION.dll:Version Checking and File Installation Libraries:10.0.19041.546 (WinBuild.160101.0800):Microsoft Corporation
		WINHTTP.dll:Windows HTTP-Dienste:10.0.19041.2248 (WinBuild.160101.0800):Microsoft Corporation
		WINMM.dll:MCI API-DLL:10.0.19041.2248 (WinBuild.160101.0800):Microsoft Corporation
		WS2_32.dll:Windows Socket 2.0-32-Bit-DLL:10.0.19041.1081 (WinBuild.160101.0800):Microsoft Corporation
		WSOCK32.dll:Windows Socket 32-Bit DLL:10.0.19041.2248 (WinBuild.160101.0800):Microsoft Corporation
		Wldp.dll:Windows-Sperrungsrichtlinie:10.0.19041.2248 (WinBuild.160101.0800):Microsoft Corporation
		amsi.dll:Anti-Malware Scan Interface:10.0.19041.2075 (WinBuild.160101.0800):Microsoft Corporation
		bcrypt.dll:Bibliothek mit kryptografischen Primitiven von Windows:10.0.19041.2248 (WinBuild.160101.0800):Microsoft Corporation
		bcryptPrimitives.dll:Windows Cryptographic Primitives Library:10.0.19041.1415 (WinBuild.160101.0800):Microsoft Corporation
		clbcatq.dll:COM+ Configuration Catalog:2001.12.10941.16384 (WinBuild.160101.0800):Microsoft Corporation
		combase.dll:Microsoft COM für Windows:10.0.19041.2248 (WinBuild.160101.0800):Microsoft Corporation
		dbgcore.DLL:Windows Core Debugging Helpers:10.0.19041.789 (WinBuild.160101.0800):Microsoft Corporation
		dhcpcsvc.DLL:DHCP Clientdienst:10.0.19041.2248 (WinBuild.160101.0800):Microsoft Corporation
		dhcpcsvc6.DLL:DHCPv6-Client:10.0.19041.2248 (WinBuild.160101.0800):Microsoft Corporation
		fwpuclnt.dll:FWP/IPsec Benutzermodus-API:10.0.19041.2248 (WinBuild.160101.0800):Microsoft Corporation
		gdi32full.dll:GDI Client DLL:10.0.19041.2130 (WinBuild.160101.0800):Microsoft Corporation
		java.dll:OpenJDK Platform binary:17.0.3.0:Microsoft
		javaw.exe:OpenJDK Platform binary:17.0.3.0:Microsoft
		jemalloc.dll
		jimage.dll:OpenJDK Platform binary:17.0.3.0:Microsoft
		jli.dll:OpenJDK Platform binary:17.0.3.0:Microsoft
		jna7641697183013218739.dll:JNA native library:6.1.2:Java(TM) Native Access (JNA)
		jsvml.dll:OpenJDK Platform binary:17.0.3.0:Microsoft
		jvm.dll:OpenJDK 64-Bit server VM:17.0.3.0:Microsoft
		kernel.appcore.dll:AppModel API Host:10.0.19041.546 (WinBuild.160101.0800):Microsoft Corporation
		lwjgl.dll
		management.dll:OpenJDK Platform binary:17.0.3.0:Microsoft
		management_ext.dll:OpenJDK Platform binary:17.0.3.0:Microsoft
		mdnsNSP.dll:Bonjour Namespace Provider:3,1,0,1:Apple Inc.
		msvcp140.dll:Microsoft® C Runtime Library:14.29.30139.0 built by: vcwrkspc:Microsoft Corporation
		msvcp_win.dll:Microsoft® C Runtime Library:10.0.19041.789 (WinBuild.160101.0800):Microsoft Corporation
		msvcrt.dll:Windows NT CRT DLL:7.0.19041.546 (WinBuild.160101.0800):Microsoft Corporation
		mswsock.dll:Microsoft Windows Sockets 2.0-Dienstanbieter:10.0.19041.2248 (WinBuild.160101.0800):Microsoft Corporation
		napinsp.dll:E-Mail-Namenshimanbieter:10.0.19041.2248 (WinBuild.160101.0800):Microsoft Corporation
		ncrypt.dll:Windows NCrypt-Router:10.0.19041.2248 (WinBuild.160101.0800):Microsoft Corporation
		net.dll:OpenJDK Platform binary:17.0.3.0:Microsoft
		nio.dll:OpenJDK Platform binary:17.0.3.0:Microsoft
		ntdll.dll:DLL für NT-Layer:10.0.19041.2248 (WinBuild.160101.0800):Microsoft Corporation
		perfos.dll:DLL für Windows-Systemleistungsobjekte:10.0.19041.2248 (WinBuild.160101.0800):Microsoft Corporation
		pnrpnsp.dll:PNRP-Namespaceanbieter:10.0.19041.2248 (WinBuild.160101.0800):Microsoft Corporation
		profapi.dll:User Profile Basic API:10.0.19041.844 (WinBuild.160101.0800):Microsoft Corporation
		rasadhlp.dll:Remote Access AutoDial Helper:10.0.19041.546 (WinBuild.160101.0800):Microsoft Corporation
		rsaenh.dll:Microsoft Enhanced Cryptographic Provider:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		sechost.dll:Host for SCM/SDDL/LSA Lookup APIs:10.0.19041.1 (WinBuild.160101.0800):Microsoft Corporation
		shlwapi.dll:Shell Light-weight-Hilfsprogrammbibliothek:10.0.19041.2248 (WinBuild.160101.0800):Microsoft Corporation
		sunmscapi.dll:OpenJDK Platform binary:17.0.3.0:Microsoft
		ucrtbase.dll:Microsoft® C Runtime Library:10.0.19041.789 (WinBuild.160101.0800):Microsoft Corporation
		vcruntime140_1.dll:Microsoft® C Runtime Library:14.29.30139.0 built by: vcwrkspc:Microsoft Corporation
		verify.dll:OpenJDK Platform binary:17.0.3.0:Microsoft
		win32u.dll:Win32u:10.0.19041.2251 (WinBuild.160101.0800):Microsoft Corporation
		windows.storage.dll:Microsoft WinRT Storage-API:10.0.19041.2248 (WinBuild.160101.0800):Microsoft Corporation
		winrnr.dll:LDAP RnR Provider DLL:10.0.19041.546 (WinBuild.160101.0800):Microsoft Corporation
		wshbth.dll:Windows Sockets Helper DLL:10.0.19041.546 (WinBuild.160101.0800):Microsoft Corporation
		zip.dll:OpenJDK Platform binary:17.0.3.0:Microsoft
Stacktrace:
	at net.minecraft.client.main.Main.method_44604(Main.java:205)
	at net.minecraft.client.main.Main.main(Main.java:51)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:461)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)

-- System Details --
Details:
	Minecraft Version: 1.19.2
	Minecraft Version ID: 1.19.2
	Operating System: Windows 10 (amd64) version 10.0
	Java Version: 17.0.3, Microsoft
	Java VM Version: OpenJDK 64-Bit Server VM (mixed mode), Microsoft
	Memory: 2829856992 bytes (2698 MiB) / 3590324224 bytes (3424 MiB) up to 17179869184 bytes (16384 MiB)
	CPUs: 12
	Processor Vendor: AuthenticAMD
	Processor Name: AMD Ryzen 5 3600 6-Core Processor              
	Identifier: AuthenticAMD Family 23 Model 113 Stepping 0
	Microarchitecture: unknown
	Frequency (GHz): 3.59
	Number of physical packages: 1
	Number of physical CPUs: 6
	Number of logical CPUs: 12
	Graphics card #0 name: NVIDIA GeForce GTX 1080
	Graphics card #0 vendor: NVIDIA (0x10de)
	Graphics card #0 VRAM (MB): 4095.00
	Graphics card #0 deviceId: 0x1b80
	Graphics card #0 versionInfo: DriverVersion=31.0.15.2756
	Memory slot #0 capacity (MB): 8192.00
	Memory slot #0 clockSpeed (GHz): 2.13
	Memory slot #0 type: DDR4
	Memory slot #1 capacity (MB): 16384.00
	Memory slot #1 clockSpeed (GHz): 2.13
	Memory slot #1 type: DDR4
	Memory slot #2 capacity (MB): 8192.00
	Memory slot #2 clockSpeed (GHz): 2.13
	Memory slot #2 type: DDR4
	Memory slot #3 capacity (MB): 8192.00
	Memory slot #3 clockSpeed (GHz): 2.13
	Memory slot #3 type: DDR4
	Virtual memory max (MB): 50387.94
	Virtual memory used (MB): 24856.28
	Swap memory total (MB): 9504.40
	Swap memory used (MB): 8.27
	JVM Flags: 9 total; -XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump -Xss1M -Xmx16G -XX:+UnlockExperimentalVMOptions -XX:+UseG1GC -XX:G1NewSizePercent=20 -XX:G1ReservePercent=20 -XX:MaxGCPauseMillis=50 -XX:G1HeapRegionSize=32M
	Fabric Mods: 
		fabric-api: Fabric API 0.69.0+1.19.2
			fabric-api-base: Fabric API Base 0.4.13+0d0f210290
			fabric-api-lookup-api-v1: Fabric API Lookup API (v1) 1.6.12+93d8cb8290
			fabric-biome-api-v1: Fabric Biome API (v1) 9.0.20+0d0f210290
			fabric-block-api-v1: Fabric Block API (v1) 1.0.1+12bfe4ea90
			fabric-blockrenderlayer-v1: Fabric BlockRenderLayer Registration (v1) 1.1.22+c6af733c90
			fabric-client-tags-api-v1: Fabric Client Tags 1.0.3+b35fea8390
			fabric-command-api-v1: Fabric Command API (v1) 1.2.13+f71b366f90
			fabric-command-api-v2: Fabric Command API (v2) 2.1.9+93d8cb8290
			fabric-commands-v0: Fabric Commands (v0) 0.2.30+df3654b390
			fabric-containers-v0: Fabric Containers (v0) 0.1.38+df3654b390
			fabric-content-registries-v0: Fabric Content Registries (v0) 3.4.4+6064db2a90
			fabric-convention-tags-v1: Fabric Convention Tags 1.2.0+27265b3490
			fabric-crash-report-info-v1: Fabric Crash Report Info (v1) 0.2.7+aeb40ebe90
			fabric-data-generation-api-v1: Fabric Data Generation API (v1) 5.3.3+ad31814d90
			fabric-dimensions-v1: Fabric Dimensions API (v1) 2.1.33+0d0f210290
			fabric-entity-events-v1: Fabric Entity Events (v1) 1.5.2+9244241690
			fabric-events-interaction-v0: Fabric Events Interaction (v0) 0.4.31+6eb9a2a190
			fabric-events-lifecycle-v0: Fabric Events Lifecycle (v0) 0.2.31+df3654b390
			fabric-game-rule-api-v1: Fabric Game Rule API (v1) 1.0.23+c6af733c90
			fabric-item-api-v1: Fabric Item API (v1) 1.6.1+fa140d5990
			fabric-item-groups-v0: Fabric Item Groups (v0) 0.3.36+9244241690
			fabric-key-binding-api-v1: Fabric Key Binding API (v1) 1.0.23+aaaf9d3390
			fabric-keybindings-v0: Fabric Key Bindings (v0) 0.2.21+df3654b390
			fabric-lifecycle-events-v1: Fabric Lifecycle Events (v1) 2.2.2+9244241690
			fabric-loot-api-v2: Fabric Loot API (v2) 1.1.10+83a8659290
			fabric-loot-tables-v1: Fabric Loot Tables (v1) 1.1.13+9e7660c690
			fabric-message-api-v1: Fabric Message API (v1) 5.0.5+93d8cb8290
			fabric-mining-level-api-v1: Fabric Mining Level API (v1) 2.1.21+33fbc73890
			fabric-models-v0: Fabric Models (v0) 0.3.19+c6af733c90
			fabric-networking-api-v1: Fabric Networking API (v1) 1.2.8+3b1d070f90
			fabric-networking-v0: Fabric Networking (v0) 0.3.25+df3654b390
			fabric-object-builder-api-v1: Fabric Object Builder API (v1) 4.1.3+06e0e25190
			fabric-particles-v1: Fabric Particles (v1) 1.0.12+79adfe0a90
			fabric-registry-sync-v0: Fabric Registry Sync (v0) 0.9.29+9244241690
			fabric-renderer-api-v1: Fabric Renderer API (v1) 1.1.0+e7aeff4590
			fabric-renderer-indigo: Fabric Renderer - Indigo 0.6.17+aeb40ebe90
			fabric-renderer-registries-v1: Fabric Renderer Registries (v1) 3.2.22+df3654b390
			fabric-rendering-data-attachment-v1: Fabric Rendering Data Attachment (v1) 0.3.17+0d0f210290
			fabric-rendering-fluids-v1: Fabric Rendering Fluids (v1) 3.0.9+c6af733c90
			fabric-rendering-v0: Fabric Rendering (v0) 1.1.25+df3654b390
			fabric-rendering-v1: Fabric Rendering (v1) 1.11.1+73145abb90
			fabric-resource-conditions-api-v1: Fabric Resource Conditions API (v1) 2.1.1+aae9039d90
			fabric-resource-loader-v0: Fabric Resource Loader (v0) 0.8.2+574a764890
			fabric-screen-api-v1: Fabric Screen API (v1) 1.0.30+b0bc9a6790
			fabric-screen-handler-api-v1: Fabric Screen Handler API (v1) 1.3.4+1cc24b1b90
			fabric-sound-api-v1: Fabric Sound API (v1) 1.0.1+c4f28df590
			fabric-textures-v0: Fabric Textures (v0) 1.0.22+aeb40ebe90
			fabric-transfer-api-v1: Fabric Transfer API (v1) 2.1.3+9d081d6490
			fabric-transitive-access-wideners-v1: Fabric Transitive Access Wideners (v1) 1.3.2+42d99c3290
		fabricloader: Fabric Loader 0.14.10
		java: OpenJDK 64-Bit Server VM 17
		minecraft: Minecraft 1.19.2
		snowy: Snowy 1.0-SNAPSHOT
	Launched Version: fabric-loader-0.14.10-1.19.2
	Backend library: LWJGL version 3.3.1 SNAPSHOT
	Backend API: Unknown
	Window size: <not initialized>
	GL Caps: Using framebuffer using OpenGL 3.2
	GL debug messages: <disabled>
	Using VBOs: Yes
	Is Modded: Definitely; Client brand changed to 'fabric'
	Type: Client (map_client.txt)
	CPU: <unknown>

**Comment by warjort** [comment]
````
Caused by: java.lang.ClassNotFoundException: net.minecraft.client.gui.screen.Screen
at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
````

Looks like you are trying to run the deobfuscated development version of your mod in an obfuscated environment?

https://fabricmc.net/wiki/tutorial:mappings#mappings
